doc/snippets/Microsoft.Data.SqlClient/ActiveDirectoryAuthenticationProvider.xml

```diff
@@ -0,0 +1,82 @@
+    </ctor2>
+    <AcquireTokenAsync>
+      <param name="parameters">The Active Directory authentication parameters passed by the driver to authentication providers.</param>
+      <summary>Acquires a security token from the authority.</summary>
+      <returns>Represents an asynchronous operation that returns the AD authentication token.</returns>
+    </AcquireTokenAsync>
+    <SetDeviceCodeFlowCallback>
+      <param name="deviceCodeFlowCallbackMethod">The callback method to be used by driver when performing 'Active Directory Device Code Flow' authentication.</param>
+      <summary>Sets callback method that overrides driver's default implementation to process result when performing 'Active Directory Device Code Flow' authentication.</summary>
+    </SetDeviceCodeFlowCallback>
+    <SetParentActivityOrWindowFunc>
+      <param name="parentActivityOrWindowFunc">The parent as an object, in order to be used from shared NetStandard assemblies.</param>
+      <summary>Sets a reference to the ViewController (if using Xamarin.iOS), Activity (if using Xamarin.Android) IWin32Window or IntPtr (if using .Net Framework). Used for invoking the browser for Active Directory Interactive authentication.</summary>
+      <remarks>Mandatory only on Android to be set. See https://aka.ms/msal-net-android-activity for further documentation and details.</remarks>
+    </SetParentActivityOrWindowFunc>
+    <SetIWin32WindowFunc>
+      <param name="iWin32WindowFunc">A function to return the current window.</param>
+      <summary>Sets a reference to the current <see cref="T:System.Windows.Forms.IWin32Window" /> that triggers the browser to be shown. Used to center the browser that pop-up onto this window."</summary>
+    </SetIWin32WindowFunc>
+    <SetCustomWebUI>
+      <param name="customWebUi">Customer implementation for the Web UI</param>
+      <summary>Sets a custom Web UI that will let the user sign-in with Azure AD, present consent if needed, and get back the authorization code. Applicable when working with Active Directory Interactive authentication.</summary>
+    </SetCustomWebUI>
+    <BeforeLoad>
+      <param name="authentication">The authentication method.</param>
+      <summary>This method is called immediately before the provider is added to SQL drivers registry. </summary>
+      <remarks>Avoid performing long-waiting tasks in this method, since it can block other threads from accessing the provider registry.</remarks>
+    </BeforeLoad>
+    <BeforeUnload>
+      <param name="authentication">The authentication method.</param>
+      <summary>This method is called immediately before the provider is removed from the SQL drivers registry. </summary>
+      <remarks>For example, this method is called when a different provider with the same authentication method overrides this provider in the SQL drivers registry. Avoid performing long-waiting task in this method, since it can block other threads from accessing the provider registry.</remarks>
```

Comment:
```suggestion
      <remarks>For example, this method is called when a different provider with the same authentication method overrides this provider in the authentication provider registry. Avoid performing long-waiting task in this method, since it can block other threads from accessing the provider registry.</remarks>
```